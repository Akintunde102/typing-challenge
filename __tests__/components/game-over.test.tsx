import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import GameOver from '../../components/game-over';


describe('Game Over Component', () => {


    describe("when presentStage is not set to -1", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<GameOver scores={{ speed: 2, points: 20 }} presentStage={0} />)
        })

        it('does not renders a text', () => {
            expect(renderedComponent.container).toBeEmptyDOMElement();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })

    })


    describe("when presentStage is set to -1", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<GameOver scores={{ speed: 2, points: 20 }} presentStage={-1} />)
        })

        it('renders end of challenge', () => {
            const text = screen.getByText("Typing Challenge Has Ended");
            expect(text).toBeInTheDocument()
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        });
    });






})