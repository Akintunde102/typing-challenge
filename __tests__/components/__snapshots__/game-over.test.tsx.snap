// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Over Component when presentStage is not set to -1 matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Game Over Component when presentStage is set to -1 matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
      >
        <div
          class="section"
        >
          <h1
            class="title"
          >
            Typing Challenge Has Ended
          </h1>
        </div>
        <div
          class="grid"
        >
          <div
            class="card"
          >
            <span
              class="score"
            >
                  Points: 
              20
            </span>
          </div>
          <div
            class="card"
          >
            <span
              class="score"
            >
              Speed: 
              2
               words/minute
            </span>
          </div>
          <h1
            class="bigButton"
          >
            Start Again
          </h1>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
    >
      <div
        class="section"
      >
        <h1
          class="title"
        >
          Typing Challenge Has Ended
        </h1>
      </div>
      <div
        class="grid"
      >
        <div
          class="card"
        >
          <span
            class="score"
          >
                Points: 
            20
          </span>
        </div>
        <div
          class="card"
        >
          <span
            class="score"
          >
            Speed: 
            2
             words/minute
          </span>
        </div>
        <h1
          class="bigButton"
        >
          Start Again
        </h1>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
